components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -69,14 +69,56 @@ AIChatTabHelper::AIChatTabHelper(
       base::BindRepeating(
           &AIChatTabHelper::OnPermissionChangedAutoGenerateQuestions,
           weak_ptr_factory_.GetWeakPtr()));
+  // TODO(petemill): AIChatCredential manager could be singleton since it uses
+  // local state prefs.
   credential_manager_ = std::make_unique<ai_chat::AIChatCredentialManager>(
       skus_service_getter, local_state_prefs);
 
-  // Engines and model names are be selectable
-  // per conversation, not static.
-  // Start with default.
-  // TODO(petemill): Default should come from pref value.
-  model_key_ = features::kAIModelKey.Get();
+  // Engines and model names are selectable per conversation, not static.
+  // Start with default from pref value but only if user set. We can't rely on
+  // actual default pref value since we should vary if user is premium or not.
+  // TODO(petemill): When we have an event for premium status changed, and a
+  // profile service for AIChat, then we can call
+  // |pref_service_->SetDefaultPrefValue| when the user becomes premium. With
+  // that, we'll be able to simply call GetString(prefs::kDefaultModelKey) and
+  // not vary on premium status.
+  if (!pref_service_->GetUserPrefValue(prefs::kDefaultModelKey)) {
+    credential_manager_->GetPremiumStatus(base::BindOnce(
+        [](AIChatTabHelper* instance, mojom::PremiumStatus status) {
+          if (!instance) {
+            // WebContents destroyed
+            return;
+          }
+          if (status == mojom::PremiumStatus::Inactive) {
+            // Not premium
+            return;
+          }
+          // Use default premium model for this instance
+          instance->ChangelModel(kModelsPremiumDefaultKey);
+          // Make sure default model reflects premium status
+          const auto* current_default =
+              instance->pref_service_
+                  ->GetDefaultPrefValue(prefs::kDefaultModelKey)
+                  ->GetIfString();
+
+          if (current_default && *current_default != kModelsPremiumDefaultKey) {
+            instance->pref_service_->SetDefaultPrefValue(
+                prefs::kDefaultModelKey, base::Value(kModelsPremiumDefaultKey));
+          }
+        },
+        base::Unretained(this)));
```

Comment:
Unretained is fine here IMO: 1) can't use WeakPtr because in ctor, 2) I included a check on `!instance` in the function body.

Reply:
> Unretained is fine here IMO: 1) can't use WeakPtr because in ctor, 2) I included a check on !instance in the function body.

the `!instance` check does nothing here, because you just pass a raw pointer. It will always have a value, even if an object is destroyed.

This is safe for a different reason: `credential_manager_` is a `unique_ptr` member of the class, `AIChatCredentialManager` internally uses `weak_ptr_factory_.GetWeakPtr()` when a callback is postponed.

Reply:
@goodov Thanks, but how about the fact that AIChatCredentialManager passes the callback to the skus service, which is longer lived? `skus_service_->CredentialSummary`

Reply:
@petemill the fact that the skus service is longer lived would make it a potential problem because the callback could happen after the tab helper is destroyed

Reply:
As discussed, the TabHelper owns the CredentialManager, which owns the mojo binding to skus service - the mechanism the async call is made via. So I think we're ok. I've added a code comment.

Reply:
> @goodov Thanks, but how about the fact that AIChatCredentialManager passes the callback to the skus service, which is longer lived? `skus_service_->CredentialSummary`

you mean this code?
```
  skus_service_->CredentialSummary(
      leo_sku_domain,
      base::BindOnce(&AIChatCredentialManager::OnCredentialSummary,
                     weak_ptr_factory_.GetWeakPtr(), std::move(callback),
                     leo_sku_domain));
```

here you don't pass it to `CredentialSummary`, but Bind it to `&AIChatCredentialManager::OnCredentialSummary` callback which is NOT be executed if the weak ptr returned by `weak_ptr_factory_` is invalidated.

and as you also mentioned, `skus_service_` is `mojo::Remote` which means it won't call callbacks if it's destroyed.

Reply:
thanks @goodov - looks like we're ok then